### ISSUE TITLE
Jump

### ISSUE THREAD

**Issue by arnaudpourbaix** [ZerdBG3/RAW#193]
Jump is one of the thing I want to fix in this game.
I am using this calculator: https://fexlabs.com/5ejump/

Now that Jump distance is based on movement, it affects Enhanced Jump.
With 9 strength, I can jump about 4.5m, and about 8m with buff.
With 16 strength, I can jump about 9m with or without buff.

Anyway, this is not as it should be in PnP:
With 9 strength, I should be able to jump up to 2.7m. So it should be 8.1 with buff
With 16 strength, I should be able to jump up to 5.2m. So it should be 15.6 with buff.

Mod combatactions found a way to affect enhanced jump by adding a spell icon when active.

I have played a bit with use cost:
`Movement:Distance*1.7 `-> My 16 strength can now jump about 5.2m, which is perfect. But my 9 str can do the same distance !
I need to find out how strength is calculated. I think it is related to: `data "AddRangeFromAbility" "Strength,1"`

Additionnally, there is also that radius circle, currently wrong about possible jump distance.
I suppose it might be related to: `data "TargetRadius" "4.5"`

I will make further tests to see how it can be improved.
Maybe your current implementation is acceptable, as high strength characters won't be benefit from enhanced leap but it will help low str characters ?

**Comment by arnaudpourbaix** [comment]
I have tried a few things this morning and I came up with this:
```
new entry "Projectile_Jump"
type "SpellData"
data "SpellType" "Projectile"
using "Projectile_Jump"
data "UseCosts" "Movement:3;Movement:Distance"
data "TargetRadius" "2"

new entry "LONG_JUMP"
type "StatusData"
data "StatusType" "BOOST"
using "LONG_JUMP"
data "Boosts" "JumpMaxDistanceMultiplier(3);UnlockSpell(Projectile_Jump_Enhanced)"

new entry "Projectile_Jump_Enhanced"
type "SpellData"
using "Projectile_Jump"
data "UseCosts" "Movement:3;Movement:Distance*0.33"
```

Normal jump is really close to PnP. So, if you have 17 str, you can make 1 jump: 5m then move 1m.
If you cast Enhanced Leap, Jump now cost 33% movement so you can finally jump 15m with 17 str. However, it does screw movement left as in my case, you can still move 1m after this 15m jump. Or maybe, you can't jump more than your movement points ?

Finally, there is also Athlete feature that need fixing. It should not change jump distance, but it should reduce movement cost to 1.5m (from 3m). I don't know how to do this right now. I need to replace jump entry by a new one, and same thing for Enhanced Leap. Doesn't seem easy ! Anyway, removing triple jump for Athlete must be done, even without this change.

**Comment by FelipeRenault** [comment]
> Long Jump. When you make a long jump, you cover a number of feet up to your Strength score if you move at least 10 feet on foot immediately before the jump. When you make a standing long jump, you can leap only half that distance. Either way, each foot you clear on the jump costs a foot of movement.

I don't think you can jump 15m with 17 str. It says you can long jump a number of feet up to your strength score. So 17 strength allows you to long jump up to 17 feet, which is about 5.1m.
It also states that each foot jumped costs a foot of movement, so distance is 1:1 when jumping. You can't jump further than your remaining movement

**Comment by arnaudpourbaix** [comment]
You are right, I've just read a comment from Jeremy Crawford that confirmed you can't jump farther than your movement speed.
You can extend your movement speed with dash to jump farther.

However, you still need to spend 3 meters to make a long jump. And High Jump is more limitated.

I am testing this:
```
new entry "Projectile_Jump"
type "SpellData"
data "SpellType" "Projectile"
using "Projectile_Jump"
data "UseCosts" "Movement:3;Movement:Distance"

new entry "Projectile_High_Jump"
type "SpellData"
data "SpellType" "Projectile"
using "Projectile_Jump"
data "AddRangeFromAbility" "Strength,0.5"
data "TargetRadius" "0.6"
data "MinJumpDistance" "0.6"
data "Trajectories" "10e3e9e2-82b1-16be-80c3-bba883d01b48,e528a4bd-bc17-d0d1-188c-dbe1d4eda1f0"
data "DisplayName" "e82d6bbb-2605-4800-8555-aca1a242d7e6"

new entry "Projectile_Long_Jump"
type "SpellData"
data "SpellType" "Projectile"
using "Projectile_Jump"
data "AddRangeFromAbility" "Strength,1"
data "TargetRadius" "2"
data "MinJumpDistance" "1"
data "Trajectories" "9d893642-654d-41a4-aaad-81f9dc588941"
data "DisplayName" "a8ce1c98-ccce-4dfe-86d9-de984cce3a03"
```

Trajectories are not working well, but you can still decide to use High or Long jump.
For these changes, you can't enter Emerald Grove with Enhanced High Jump (only 6m). But you can do it with Enhanced Long Jump if you use dash before. You shouldn't be able to do it, but I can't fix trajectories.

**Comment by FelipeRenault** [comment]
One of the reasons I didn't try to perfectly match jumping distance in the mod is because the map itself is designed to match their implementation of jump. So by tweaking the max distance, it might result in making areas inaccessible early in the game without spells, or make a combat that would allow you to position better way harder (I don't think it should be easy to position oneself with a single jump, but all the maps and terrains were designed with the vanilla jump in mind, so even if they've made an "easy to position" terrain for a combat, it would possibly become a hard terrain)

**Comment by FelipeRenault** [comment]
By the way, I thought of one way to implement long jump. Instead of making the distance part of the movement cost, you'd split the spell in two parts: first a rush spell that dashes a fixed distance (if that's possible) and costs 3m of movement. After using the rush spell, it'd unlock a jump spell with the enhanced distance which only costs movement. The distance is based on the strength as per usual. I'm not sure if it's possible to enforce a direction for a jump spell, though. It'd be cool if that was possible

This is a very convoluted solution for the direction enforcing (and it might require scripting), but it might work: after dashing with the rush spell, it'd automatically cast a zone spell with a specified cone angle (maybe 60º) and a very long range. That would create an invisible surface in which is a requirement for the enhanced jump targeting. This would ensure that after dashing, the enhanced jump is used in the general direction of the dash

**Comment by arnaudpourbaix** [comment]
It seems to be very large work but small result.
`data "UseCosts" "Movement:3;Movement:Distance"`
This one works just as fine, you don't need to move but you will pay those 3 meters. Sometimes, you can't move enough to jump after, but these situations are not very common.

I tested broken bridge in act 1 and without buff, low strength can't make that jump. So, it should apply on other locations as well. They could make it with enhanced leap though. And it is not difficult to get Enhanced Leap as a ritual spell or potion. Casters might get fly spell later on.

Actually, you can make everything by jumping. I have never use enhanced leap in act 1 because usually, basic jump is enough to get the job done. For battle, enhanced leap on melee is helpfull but not mandatory. 
I think they might have deliberately made big jumps to make the game more simple, without needing to use effects like enhanced leap.

But nevermind, I understand that many players wouldn't accept this. I will add this change to my own personal mod :)

**Comment by FelipeRenault** [comment]
But won't `data "UseCosts" "Movement:3;Movement:Distance"` make it so you pay 3m to use plus the distance, including the 3m again? For example, by the rules, to cross a 5m gap, you'd need to run 3m and then jump 5m, costing you 8m of movement. Without me testing it, it feels like you'd jump 8m (simulating a running start of 3m and the 5m jump), but it'd cost you 3m flat plus the distance jumped, which would be 8m, thus costing you 11m total

**Comment by arnaudpourbaix** [comment]
No, let say you want to jump 5m. You will pay those 5m with Movement:distance, but you will also pay 3m with Movement:3
So in the end, you will have 8m to jump 5m. In PnP, you would have walk 3m then jump 5m.
If you remove 3m cost, you can jump more because you won't pay those 3m. Normally, you should jump half distance without running 3m first.
I have made several tests and I can confirm it behaves like I said.

It can be unfair, because if you move 3m then jump, you will loose 3m for nothing. So, maybe removing this 3m cost and assume player will always move before jumping ? In my experience, both situations happen regularly, there is no clear winner here.

Ideailly, jump should be activated if you have moved 3m since nobody would use a half distance jump. But to do this, you have to keep in mind in and out of combat situation.

**Comment by FelipeRenault** [comment]
I don't see a benefit for programming all the changes for half-distance and long jump if there's no way to enforce the running start. Simply spending the 3m will be a detriment to the player, since they will end up traversing less terrain with their movement when jumping. This solution makes jump go from being strong (in vanilla, or even in my mod) to jump being weak, when ideally, we'd meet at the "balanced" point where it's closer to the rules

**Comment by arnaudpourbaix** [comment]
Yeah, I finally agree with you. I will just retain a smaller radius of 2 instead 4.5 which makes it closer to PnP. I understand it might make some situations more difficult because you will need some way to increase jump distance.
Now for your mod, you might consider to find a value in between, maybe 3 or 3.5. Actually, with your mod, you can jump 9m with 17 str, which is considerable if you think about it.

**Comment by arnaudpourbaix** [comment]
I have just made this spreadsheet:
| Strength | Modifier | PnP | BG3 | Diff| Mod | Diff| Radius | Ability coef  | 
|----------|----------|-----|-----|-------|-----|--------|--------|---------------|
| 8        | -1       | 2,4 | 3,5 | 1,1   | 2,4 | 0,0    | 3,05   | 0,61          | 
| 9        | -1       | 2,7 | 3,5 | 0,8   | 2,4 | -0,3   | 3,05   | 0,61          | 
| 10       | 0        | 3,0 | 4,5 | 1,5   | 3,1 | 0,0    | 3,05   | 0,61          | 
| 11       | 0        | 3,4 | 4,5 | 1,1   | 3,1 | -0,3   | 3,05   | 0,61          | 
| 12       | 1        | 3,7 | 5,5 | 1,8   | 3,7 | 0,0    | 3,05   | 0,61          | 
| 13       | 1        | 4,0 | 5,5 | 1,5   | 3,7 | -0,3   | 3,05   | 0,61          | 
| 14       | 2        | 4,3 | 6,5 | 2,2   | 4,3 | 0,0    | 3,05   | 0,61          | 
| 15       | 2        | 4,6 | 6,5 | 1,9   | 4,3 | -0,3   | 3,05   | 0,61          | 
| 16       | 3        | 4,9 | 7,5 | 2,6   | 4,9 | 0,0    | 3,05   | 0,61          | 
| 17       | 3        | 5,2 | 7,5  | 2,3   | 4,9 | -0,3   | 3,05   | 0,61          |
| 18       | 4        | 5,5 | 8,5  | 3,0   | 5,5 | 0,0    | 3,05   | 0,61          |
| 19       | 4        | 5,8 | 8,5  | 2,7   | 5,5 | -0,3   | 3,05   | 0,61          |
| 20       | 5        | 6,1 | 9,5  | 3,4   | 6,1 | 0,0    | 3,05   | 0,61          |
| 21       | 5        | 6,4 | 9,5  | 3,1   | 6,1 | -0,3   | 3,05   | 0,61          |
| 22       | 6        | 6,7 | 10,5 | 3,8   | 6,7 | 0,0    | 3,05   | 0,61          |
| 23       | 6        | 7,0 | 10,5 | 3,5   | 6,7 | -0,3   | 3,05   | 0,61          |
| 24       | 7        | 7,3 | 11,5 | 4,2   | 7,3 | 0,0    | 3,05   | 0,61          |
| 25       | 7        | 7,6 | 11,5 | 3,9   | 7,3 | -0,3   | 3,05   | 0,61          |

To summarise:
- default game radius is 4.5 and strength modifier is 1. As you can see, at 20 str, you can jump 3.4m more than you should.
- ideal values are radius of 3.05 et strength modifier of 0.61. Even values are perfect, but odd ones are slighty less, this is because we should calculate with strength ability and not strength modifier.

Here is what I tested:
```
data "TargetRadius" "3.05"
data "MinJumpDistance" "2"
data "AddRangeFromAbility" "Strength,0.61"
```

**Comment by FelipeRenault** [comment]
This change I agree with a lot. So, 1 feet is approx. 0.305 meters, which is why the coeficient should be 0.61, because the game only increments the range every two points, since it uses the modifier instead of the score itself.
However, I'd suggest the base radius to be 3.2 instead of 3.05, so instead of having a `-0.3/0` differential on distance, you'd get `-0.15/+0.15`. The graphs below should visually explain the difference between each configuration:
![image](https://github.com/ZerdBG3/RAW/assets/40004649/6da4292f-0a0e-4bc9-a6b1-7ee5a3040285)
![image](https://github.com/ZerdBG3/RAW/assets/40004649/5bc1dda5-02e4-4349-8bd9-94b21b095c8f)

**Comment by arnaudpourbaix** [comment]
Precisely, 1 meter is 3,28084 feet. You can find exact number on google. I can share my google spreadsheet if you want to get a look.

**Comment by FelipeRenault** [comment]
The graph's coeficient (inclination) should be the conversion of feet to meters, so the inverse of your number: `1/3.28084`, which is around `0.305`. So we can safely use that to match the angle of the pnp graph. Therefore, `0.61` is as close as it can get.
I just altered the base range from `3.05` to `3.2` making it so the mean of the modded graph overlaps with the pnp one, instead of being offset to a lower jumping distance.

**Comment by arnaudpourbaix** [comment]
This is good. Even if most players will have even ability score, you often start with an odd value.
Please keep in mind that it will severly alter some jumps like the broken bridge in act 1. I have just tested and even with 12 str, you can't make it.
Now, you will really want to get Enhanced Leap with Gale, Druid, or Tav. You can also jump down the bridge, get prone and loose some health. You can also cross on the river side near Karlach, 8 str is enough.

**Comment by arnaudpourbaix** [comment]
Don't forget to adjuste RAW_Projectile_Jump_Thief:
`data "AddRangeFromAbility" "Dexterity,0.61"
`
